### PR TITLE
refactor: unit test amvfitter

### PR DESCRIPTION
**Pull request by felix-russo**
- Reactivate checks in the unit test of `AMVFitter` that were commented out.
- Use logger in the unit test of `AMVFitter`.

### CODE REVIEW

File: Tests/UnitTests/Core/Vertexing/AdaptiveMultiVertexFitterTests.cpp [hunk 1]
```diff
@@ -152,15 +154,11 @@ BOOST_AUTO_TEST_CASE(adaptive_multi_vertex_fitter_test) {
...
-  if (debugMode) {
-    std::cout << "All vertices in test case: " << std::endl;
-  }
   for (auto& vtx : vtxList) {
-    if (debugMode) {
-      cv++;
-      std::cout << "\t" << cv << ". vertex ptr: " << &vtx << std::endl;
-    }
+    cv++;
+    ACTS_DEBUG("\t" << cv << ". vertex ptr: " << &vtx);
```
Comment (AJPfleger) [inline Tests/UnitTests/Core/Vertexing/AdaptiveMultiVertexFitterTests.cpp:161]:
```suggestion
    ACTS_DEBUG("\t" << &vtx - vtxList.begin() + 1 << ". vertex ptr: " << &vtx);
```
You could try this approach, if you want to completely get rid of the debug-variable `cv`.

Could also work for some loops below.

File: Tests/UnitTests/Core/Vertexing/AdaptiveMultiVertexFitterTests.cpp [hunk 2]
```diff
@@ -259,44 +250,37 @@ BOOST_AUTO_TEST_CASE(adaptive_multi_vertex_fitter_test) {
...
+  ACTS_DEBUG("Vertex positions after fit of vertex 1 and 2:");
+  ACTS_DEBUG("Vtx 1, seed position:\n " << seedListCopy.at(0).fullPosition()
+                                        << "\nFitted position:\n "
+                                        << vtxList.at(0).fullPosition());
+  ACTS_DEBUG("Vtx 2, seed position:\n " << seedListCopy.at(1).fullPosition()
+                                        << "\nFitted position:\n "
+                                        << vtxList.at(1).fullPosition());
+  ACTS_DEBUG("Vtx 3, seed position:\n " << seedListCopy.at(2).fullPosition()
+                                        << "\nFitted position:\n "
+                                        << vtxList.at(2).fullPosition());
```
Comment (AJPfleger) [inline Tests/UnitTests/Core/Vertexing/AdaptiveMultiVertexFitterTests.cpp:283]:
```suggestion
for (size_t vtxIter = 0; vtxIter < 3; vtxIter++) {
    ACTS_DEBUG("Vtx " << vtxIter + 1 << ", seed position:\n " << seedListCopy.at(vtxIter).fullPosition()
                                          << "\nFitted position:\n "
                                          << vtxList.at(vtxIter).fullPosition());
}
```
Same below.

**Review by AJPfleger — APPROVED** [review]

File: Tests/UnitTests/Core/Vertexing/AdaptiveMultiVertexFitterTests.cpp [hunk 3]
```diff
@@ -54,6 +55,9 @@ namespace Test {
 using namespace Acts::UnitLiterals;
 using Acts::VectorHelpers::makeVector4;
 
+// Set up logger
+ACTS_LOCAL_LOGGER(Acts::getDefaultLogger("AMVFitterTests", Acts::Logging::INFO))
```
Comment (paulgessinger) [inline Tests/UnitTests/Core/Vertexing/AdaptiveMultiVertexFitterTests.cpp:59]: This is not actually needed anymore I think. Just do:

```suggestion
auto logger = Acts::getDefaultLogger("AMVFitterTests", Acts::Logging::INFO)
```

Comment (paulgessinger) [reply Tests/UnitTests/Core/Vertexing/AdaptiveMultiVertexFitterTests.cpp:59]: Scratch that, never mind.

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [27471833e6a53c09dadaccbc55dcb880dfe2f20c]
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6461287)!

|status|job|
|------|---|
|:red_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33842218) |
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33842147) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33841577) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33841576) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33841575) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33841574) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33841573) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33841572) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33841571) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33841570) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33841569) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33841568) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33841567) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33841566) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33841565) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33841564) |
|:red_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33841562) |
|:red_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33841561) |

**Comment by paulgessinger** [comment]
I think the output failures are bogus. Checking if they're fixed now.

**Pull request merged**